crates/client/mempool/src/lib.rs

```
@@ -0,0 +1,193 @@
+                        header: PendingHeader {
+                            parent_block_hash: block_info.block_hash,
+                            sequencer_address: **self.backend.chain_config().sequencer_address,
+                            block_timestamp: SystemTime::now()
+                                .duration_since(SystemTime::UNIX_EPOCH)
+                                .expect("Current time is before the unix timestamp")
+                                .as_secs(),
+                            protocol_version: self.backend.chain_config().latest_protocol_version,
+                            l1_gas_price: self.l1_data_provider.get_gas_prices(),
+                            l1_da_mode: self.l1_data_provider.get_da_mode(),
+                        },
+                        tx_hashes: vec![],
+                    }),
+                    inner: DeoxysBlockInner { transactions: vec![], receipts: vec![] },
+                })
+            }
+        }
+    }
+
+    pub fn accept_account_tx(&self, tx: AccountTransaction) -> Result<(), Error> {
+        // The timestamp *does not* take the transaction validation time into account.
+        let arrived_at = ArrivedAtTimestamp::now();
+
+        // Get pending block
+        let pending_block_info = self.get_or_create_pending_block()?;
+
+        // If the contract has been deployed for the same block is is invoked, we need to skip validations.
+        // NB: the lock is NOT taken the entire time the tx is being validated. As such, the deploy tx
+        //  may appear during that time - but it is not a problem.
+        let deploy_account_tx_hash = if let AccountTransaction::Invoke(tx) = &tx {
+            let mempool = self.inner.read().expect("Poisoned lock");
+            if mempool.has_deployed_contract(&tx.tx.sender_address()) {
+                Some(tx.tx_hash) // we return the wrong tx hash here but it's ok because the actual hash is unused by blockifier
```

It's strange that the blockifier takes the hash but doesn't use it. It's nice that this works right now but I think we should add a TODO to send the actual hash. If the blockifier updates later to use this hash then our code will silently fail without us noticing.

---

I think the relevant is gone in the next release anyway...

---

I didn't understand, will this code not be used in the next release?

---

I think the field is actually gone in newer blockifier or totally different